tests/dataloader_tests.cpp

```diff
@@ -58,11 +60,28 @@ BOOST_AUTO_TEST_CASE(CSVDataLoaderTest)
 BOOST_AUTO_TEST_CASE(MNISTDataLoaderTest)
 {
   DataLoader<> dataloader("mnist", true, 0.80);
+
   // Check for correct dimensions.
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 33600);
+  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_rows, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_rows, 784);
+
+
+  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_cols, 8400);
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 33600);
+  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_cols, 28000);
+
+  BOOST_REQUIRE_EQUAL(std::get<0>(dataloader.TrainSet()).n_cols, 8400);
+  BOOST_REQUIRE_EQUAL(std::get<1>(dataloader.TrainSet()).n_rows, 1);
+  BOOST_REQUIRE_EQUAL(std::get<0>(dataloader.ValidSet()).n_cols, 33600);
+  BOOST_REQUIRE_EQUAL(std::get<1>(dataloader.TrainSet()).n_rows, 1);
```

Comment:
We also need to add a comment for this block

Reply:
Will do in the next commit. Thanks.